pallets/identity/src/lib.rs

```diff
@@ -145,5 +158,20 @@ pub mod pallet {
 
             Ok(())
         }
+
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        /// Issue a signed Fennel signal on behalf of an owned identity.
+        pub fn sign_for_identity(origin: OriginFor<T>, identity_id: u32, content: Vec<u8>) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            ensure!(Self::is_identity_owned_by_sender(&who, &identity_id), Error::<T>::IdentityNotOwned);
+            let signal_id: u32 = <SignalCount<T>>::get();
+            <FennelSignal<T>>::insert(&identity_id, &signal_id, &content);
+            let new_id: u32 = signal_id.checked_add(1).ok_or(Error::<T>::StorageOverflow)?;
```

Comment:
placing `new_id` above `insert` would conform to the preference to fail before changing state

Reply:
Awesome catch! Didn't even notice that.

Reply:
Resolved in 4a034cf.